modules/ROOT/pages/6.0-release-notes-apis.adoc

```diff
@@ -0,0 +1,209 @@
+
+The dialog API’s `setData` method now uses a deep merge algorithm.
+
+Consequently, it now supports partial nested objects.
+
+[[promise]]
+==== `success` and `failure` callbacks replaced by `Promise` in multiple APIs
+
+ Previously, the AddOnManager, PluginManager, ScriptLoader, StyleSheetLoader, and ThemeManager APIs report resource loading status with `success` or `failure` callbacks.
+
+These APIs now, instead, return a `Promise`, with a value of either `resolve` or `reject`. Also, `EditorUpload`, which already returns a `Promise`, had callbacks removed.
+
+[[changed-apis]]
+=== Changed APIs
+
+[[closebutton]]
+==== `closeButton` and `timeout`
+
+* Previously, using `closeButton` to disable a _Notification’s_ close button required a `timeout` value to also be set.
++
+In {productname} 6, `closeButton` shows or hides the close button regardless of a `timeout` value being set or not set.
++
+Equivalently, setting a `timeout` value sets how long the notification presents before automatically closing, regardless of whether a close button is displayed or not.
+
+[[disabled]]
+==== `disabled` functions renamed in all {productname} APIs
+
+Previously, multiple {productname} APIs and UI classes used `disable` or `disabled` in their function or property names.
+
+In {productname} 6, these function names have been changed and configuration changes are required.
+
+. `isDisabled()` functions are now `isEnabled() functions.
+. The `enable()` and `disable()` functions are now the single `setEnabled(<state>)` function.
+. The `disable(<name>)` and `enable(<name>)` functions are now the single `setEnabled(<name>, <state>)` function in the Dialog APIs.
+
+As well, the `.disabled` property is now the `.enabled` property in all {productname} UI classes.
+
+[[dom-domutils]]
+==== `dom.DOMUtils`
+
+The `toHex` function has been removed from `dom.DOMUtils`.
+
+[[domparser]]
+==== `DomParser`
+
+The `filterNode` method has been removed from the `DomParser` API.
+
+As well, the `DomParser` API no longer uses a custom parser internally and instead uses the https://developer.mozilla.org/en-US/docs/Web/API/DOMParser[native `DOMParser` API].
+
+[[editor-setcontent]]
+==== `editor.SetContent`
+
+The `SetContent` editor event’s `content` property has been deprecated.
+
+[[env]]
+==== `Env`
+
+The following methods have been removed from the `Env` API:
+
+* `opera`
+* `webKit`
+* `ie`
+* `gecko`
+* `mac`
+* `iOS`
+* `android`
+* `desktop`
+* `contentEditable`
+* `caretAfter`
+* `range`
+* `ceFalse`
+* `experimentalShadowDom`
+* `fileApi`
+
+Note, in particular, the removal of all legacy browser detection methods.
+
+[[fire]]
+==== `fire` functions renamed in all {productname} APIs
+
+The `fire()` function in `tinymce.Editor`, `tinymce.dom.EventUtils`, `tinymce.dom.DOMUtils`, `tinymce.util.Observable` and `tinymce.util.EventDispatcher` has been renamed to `dispatch()`.
+
+`fire` has been aliased to `dispatch` but has also been marked as _deprecated_.
```

Comment:
I feel like this wording suggests fire has been deprecated, not dispatch. How about:

```suggestion
`dispatch` is now an alias to `fire` and has been marked as _deprecated_.
```

Reply:
Re-written as follows:

`fire` has been aliased to `dispatch`, so existing configurations should still work as expected.

`fire` has also, however, been marked as _deprecated_. Editing configurations to use `dispatch()` is recommended.

A useful reminder that brevity is not a virtue in and of itself. Thank you.

Also, edited equivalent material in `migration-from-5x.adoc`.

Reply:
Configurations is probably the wrong term there. `integrations` would be my suggestion instead, as it's not going to be a configuration but either the logic they've written to integration TinyMCE into the product or something like a third-party plugin.

Reply:
To quote my commit message just now:

Further edit: s/configurations/integrations/

Also edited final sentence in both sections to be straightforwardly specific: use `dispatch`; don’t use `fire`.